tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/graph_executions/graph_executions_component.ng.html

```diff
@@ -20,5 +20,40 @@
     Graph Executions ({{ numGraphExecutions }})
   </div>
 
-  <!-- TODO(cais): Add cdk-virtual-scroll-viewport for graph executions -->
+  <cdk-virtual-scroll-viewport
+    #executionsScroll
+    *ngIf="numGraphExecutions !== null && numGraphExecutions > 0"
+    itemSize="37"
+    class="graph-executions-viewport"
+    (scrolledIndexChange)="onScrolledIndexChange.emit($event)"
+  >
+    <div
+      *cdkVirtualFor="let i of graphExecutionIndices"
+      class="tensor-container"
+    >
+      <div class="tensor-item">
+        <div class="graph-execution-index">
+          {{ i }}
+        </div>
+        <div *ngIf="graphExecutionData[i]; else dataLoading">
+          <div class="tensor-name-and-op-type">
+            <div class="tensor-name">
```

Comment:
In these cases here I do want `div`, because I want them to be one separate lines (e.g., `tensor-name` and `op-type`) and be nested (e.g., `tensor-name-and-op-type` and `tensor-name`). But follow-up CLs will indeed use a lot of spans. Stay tuned.